python_modules/automation/automation/dagster_dev/commands/dg_api_record.py

```diff
@@ -0,0 +1,212 @@
+    )
+    fixture_dir = api_tests_dir / f"{domain}_tests" / "fixtures" / fixture_name
+
+    fixture_dir.mkdir(parents=True, exist_ok=True)
+    click.echo(f"🚀 Recording for {domain}.{fixture_name}: {command}")
+
+    # Ensure --json flag is present for structured output
+    if "--json" not in command:
+        click.echo(
+            f"Error: --json flag is required for GraphQL response capture: {command}", err=True
+        )
+        return False
+
+    try:
+        # Execute the dg command and capture output
+        result = subprocess.run(
+            shlex.split(command),
+            capture_output=True,
+            text=True,
+            check=False,
+            timeout=60,
+        )
+
+        # Note: cli_output.txt files are no longer created as they're not used by tests
+        # Tests use syrupy snapshots instead
+
+        # For JSON commands, also save the parsed response as a numbered JSON file
+        if result.returncode == 0:
+            try:
+                graphql_response = json.loads(result.stdout)
+                json_file = fixture_dir / "01_response.json"
+                with open(json_file, "w") as f:
+                    json.dump(graphql_response, f, indent=2, sort_keys=True)
+                click.echo(f"✅ Captured successful response for {fixture_name}")
+                click.echo(f"📄 Saved to {json_file}")
+            except json.JSONDecodeError as e:
+                click.echo(f"Warning: Command output is not valid JSON: {e}")
+                click.echo("No JSON file created")
+        else:
+            click.echo(f"⚠️  Command failed with exit code {result.returncode}")
+            if fixture_name.startswith("error_"):
+                click.echo(f"✅ Captured error response for {fixture_name}")
+            else:
+                click.echo(f"❌ Unexpected failure for {fixture_name}")
+                return False
+
+        return True
+
+    except subprocess.TimeoutExpired:
+        click.echo(f"Error: Command timed out after 60 seconds: {command}", err=True)
+        return False
+    except KeyboardInterrupt:
+        click.echo("\n⏹️  Recording interrupted", err=True)
+        return False
+    except Exception as e:
+        click.echo(f"Error executing command: {e}", err=True)
+        return False
+
+
+@click.command(name="dg-api-record")
+@click.argument("domain", required=True)
+@click.option(
+    "--fixture",
+    type=str,
+    help="Specific fixture to record (if omitted, runs all fixtures in domain)",
+)
+@click.option("--graphql", is_flag=True, help="Record GraphQL responses (Step 1)")
+def dg_api_record(domain: str, fixture: str, graphql: bool):
+    """Record GraphQL responses for API tests.
+
+    Records GraphQL responses for API tests. Can process a single fixture or all fixtures in a domain.
+
+    Args:
+        domain: API domain (e.g., 'deployment', 'run', 'asset')
+
+    Examples:
+        # Record entire domain
+        dagster-dev dg-api-record asset
```

Comment:
name here is a little confusing given that 'dagster dev' is also a user-facing thing. I don't have an obvious alternative

Reply:
`dagster-dev` an existing utility. Happy to readdress as that downside is real. See [thread](https://dagsterlabs.slack.com/archives/C047L6H0LF4/p1755190839729889)